Sasquatch/Sasquatch/ViewControllers/MSDocumentDetailsViewController.swift

```diff
@@ -3,19 +3,214 @@
   }
 
   @IBAction func backButtonClicked(_ sender: Any) {
     self.presentingViewController?.dismiss(animated: true, completion: nil)
   }
+
+  func numberOfSections(in tableView: UITableView) -> Int {
+    if documentType != userType {
+      return 1
+    } else if (documentId != nil && documentId!.isEmpty) {
+      return 2
+    }
+    return 3
+  }
+
+  func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+    if shouldDisplayProperties(in: section) {
+      return userDocumentAddPropertiesSection.tableView(tableView, numberOfRowsInSection: section)
+    } else if documentContent == nil {
+      return 1
+    } else if documentType == userType && section == 1 {
+      return 0
+    }
+    return 4
+  }
+
+  func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
+    if shouldDisplayProperties(in: indexPath.section) {
+      return userDocumentAddPropertiesSection.tableView(tableView, canEditRowAt:indexPath)
+    } else if documentType == userType && (documentContent == nil){
+      return true
+    }
+    return false
+  }
+
+  func tableView(_ tableView: UITableView, editingStyleForRowAt indexPath: IndexPath) -> UITableViewCellEditingStyle {
+    if shouldDisplayProperties(in: indexPath.section) {
+      return userDocumentAddPropertiesSection.tableView(tableView, editingStyleForRowAt: indexPath)
+    }
+    return .none
+  }
+
+  func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCellEditingStyle, forRowAt indexPath: IndexPath) {
+    if shouldDisplayProperties(in: indexPath.section) {
+      userDocumentAddPropertiesSection.tableView(tableView, commit: editingStyle, forRowAt: indexPath)
+    }
+  }
+
+  func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+    tableView.deselectRow(at: indexPath, animated: true)
+    if shouldDisplayProperties(in: indexPath.section) && userDocumentAddPropertiesSection.isInsertRow(indexPath) {
+      self.tableView(tableView, commit: .insert, forRowAt: indexPath)
+    }
+  }
+
+  func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+    if shouldDisplayProperties(in: indexPath.section) {
+      return userDocumentAddPropertiesSection.tableView(tableView, cellForRowAt:indexPath)
+    } else if documentContent != nil {
+      let cell = tableView.dequeueReusableCell(withIdentifier: "property", for: indexPath)
+      var cellText = ""
+      switch indexPath.row {
+        case 0:
+          cellText = "Document ID: \(documentContent?.documentId ?? "")"
+        break
+        case 1:
+          cellText = "Partition: \(documentContent?.partition ?? "")"
+        break
+        case 2:
+          guard let error = documentContent?.error else {
+            let formatter = DateFormatter()
+            formatter.dateFormat = "dd MMM yyyy HH:mm"
+            cellText = "Date: \(formatter.string(from: documentContent?.lastUpdatedDate ?? Date()))"
```

Comment:
Let's display text `LastUpdatedDate: ...` instead of `Date: ...`.

In the error case of nil `documentContent` or nil `lastUpdatedDate`, let's display "Unknown" instead of current time to avoid confusion.